src/main/java/io/orangebeard/listener/OrangebeardTestSystemListener.java

```diff
@@ -1,141 +1,159 @@
+import lombok.SneakyThrows;
 
+import javax.annotation.Nullable;
 import java.io.Closeable;
+import java.io.File;
 import java.io.IOException;
-import java.time.Instant;
-import java.util.*;
+import java.net.InetAddress;
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Properties;
+import java.util.Set;
+import java.util.UUID;
 import java.util.stream.Collectors;
 
-import static java.lang.String.format;
+import static io.orangebeard.listener.helper.TestPageHelper.getTestName;
 import static java.util.Objects.requireNonNull;
 
 @NoArgsConstructor
 public class OrangebeardTestSystemListener implements TestSystemListener, Closeable {
+    private String PROPERTY_FILE_NAME = "orangebeard.properties";
+    private static final String PROP_ROOT_PATH = "fitnesseroot.path";
 
-    private static final String PROP_LAUNCHNAME = "rp.launch";
-    private static final String PROP_AUGMENTED_ENDPOINT = "at.augmented-testing.endpoint";
-    private static final String PROP_AUGMENTED_TOKEN = "at.augmented-testing.token";
-    private static final String AUGMENTED_TESTING_TOKEN = "AUGMENTED_TESTING_TOKEN";
+    private OrangebeardClient orangebeard;
+    private final OrangebeardProperties orangebeardProperties = new OrangebeardProperties();
+    private ToolchainRunningContext runContext;
 
-    private static final String PROP_TAGS = "rp.tags";
-    private static final String PROP_ATTRIBUTES = "rp.attributes";
-    private final org.slf4j.Logger logger = LoggerFactory.getLogger(OrangebeardTestSystemListener.class);
-    private final String PROPERTY_FILE_NAME = "reportportal.properties";
-    private OrangebeardLogger orangebeardLogger;
-    private ReportPortal reportPortal;
-    private Launch launch;
-    private ToolchainRunningContext context = new ToolchainRunningContext();
+    private String rootPath = getFitnesseRootPath();
     private final OrangebeardTableLogParser htmlChunkParser = new OrangebeardTableLogParser();
-    private Maybe<String> launchId;
+    private OrangebeardLogger orangebeardLogger;
+
+    private boolean local = false;
+
 
+    public OrangebeardTestSystemListener(@Nullable String propertyFile) {
+        if (propertyFile != null) {
+            PROPERTY_FILE_NAME = propertyFile;
+        }
+        rootPath = getFitnesseRootPath();
+    }
 
-    public OrangebeardTestSystemListener(OrangebeardLogger orangebeardLogger, ReportPortal reportPortal, ToolchainRunningContext context) {
-        this.orangebeardLogger = orangebeardLogger;
-        this.reportPortal = reportPortal;
-        this.context = context;
+    public OrangebeardTestSystemListener(@Nullable String propertyFile, String rootPath) {
+        if (propertyFile != null) {
+            PROPERTY_FILE_NAME = propertyFile;
+        }
+        this.rootPath = rootPath;
+    }
+
+    public OrangebeardTestSystemListener(String propertyFile, boolean local) {
+        this(propertyFile);
```

Comment:
ik zou hier ook gewoon propertyfilename direct instantietieren ipv this. alle constructors werken nu verschillend ;-)